.github/workflows/update-artifacts.yml

```diff
@@ -0,0 +1,62 @@
+name: Update Artifacts
+
+on:
+  workflow_dispatch: # Allows manual triggering
+  schedule:
+    - cron: "0 0 * * 1" # weekly on Monday at 00:00 UTC
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Renew Artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      # We run this (up-to-date) node locally to fetch metadata from it for the artifacts
+      - name: Use substrate-node binary
+        uses: ./.github/workflows/actions/use-substrate
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@3cf7f8cc28d1b4e7d01e3783be10a97d55d483c8 # v2.7.1
+
+      # Needs to be built before, so that it starts quickly when called by artifacts script.
+      - name: Build the CLI tool)
+        run: cargo build --bin subxt
+
+      # Runs two processes asynchronously:
+      # - one for the substrate node, timing out after 120s.
+      # - and one to fetch the metadata from it with the `scripts/artifacts.sh` script. This waits for 30s first to give the node enough time to start up.
+      - name: Fetch Artifacts
+        run: |
+          { timeout 120s substrate-node --dev --tmp; } & { sleep 30s && ./scripts/artifacts.sh; }
```

Comment:
I would prefer if you could write a simple script/binary to actually is using the `substrate node runner` that actually parses the assigned rpc port and ensures that node has been started so we don't need the sleeps here before lauching `./scripts/artifacts.sh` here.

Recall, the substrate may assign a random port if 9944 is already used and then this will not work